File: apps/codehelper/src/lib/types/provider.ts
import type { AppMode } from '$lib/types/mode';

export interface ProviderStateDto {
	mode: AppMode;
	state: string;
	detail: string | null;
	supportsTools: boolean;
	supportsUndo: boolean;
}

export interface ToolDefinitionDto {
	name: string;
	description: string;
	inputSchema: unknown;
}

export interface ToolExecutionResultDto {
	name: string;
	ok: boolean;
	summary: string;
	payload: unknown;
}

export interface ModeStatusDto {
	mode: AppMode;
	engineReady: boolean;
	providerState: ProviderStateDto;
	availableTools: ToolDefinitionDto[];
	lastError: string | null;
}

File: crates/smolpc-assistant-types/src/provider.rs
use crate::mode::AppMode;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ProviderStateDto {
    pub mode: AppMode,
    pub state: String,
    pub detail: Option<String>,
    pub supports_tools: bool,
    pub supports_undo: bool,
}

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct ToolDefinitionDto {
    pub name: String,
    pub description: String,
    pub input_schema: serde_json::Value,
}

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct ToolExecutionResultDto {
    pub name: String,
    pub ok: bool,
    pub summary: String,
    pub payload: serde_json::Value,
}

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct ModeStatusDto {
    pub mode: AppMode,
    pub engine_ready: bool,
    pub provider_state: ProviderStateDto,
    pub available_tools: Vec<ToolDefinitionDto>,
    pub last_error: Option<String>,
}
